homeassistant/components/alert/__init__.py

```diff
@@ -252,11 +257,18 @@ async def end_alerting(self):
     async def _schedule_notify(self):
         """Schedule a notification."""
         delay = self._delay[self._next_delay]
-        next_msg = now() + delay
-        self._cancel = event.async_track_point_in_time(
-            self.hass, self._notify, next_msg
-        )
-        self._next_delay = min(self._next_delay + 1, len(self._delay) - 1)
+        # #4851 - Repeating with a delay of zero causes infinite looping when it is the last
+        # delay specified, skip this case and log a warning.
+        if delay.total_seconds() <= 0.0 and self._next_delay == len(self._delay) - 1:
+            _LOGGER.warning(
+                f"Alert: {self._attr_name} - Repeat value [{delay.total_seconds()}] must be a positive non-zero number.  Please update your configuration"
+            )
```

Comment:
Since you have blocked the condition on the inputs, perhaps an assert is more appropriate here now.

Reply:
Agree, since I have now blocked the input and it has a good error message, I don't think we need this change at all.  I've removed that piece of code.